lib/flowViewport/sceneIndex/fvpParentDataModifierSceneIndex.h

```diff
@@ -0,0 +1,97 @@
+//
+// Copyright 2023 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#ifndef FLOW_VIEWPORT_SCENEINDEX_PARENT_DATA_MODIFIER_SCENE_INDEX_H
+#define FLOW_VIEWPORT_SCENEINDEX_PARENT_DATA_MODIFIER_SCENE_INDEX_H
+
+//Local headers
+#include "flowViewport/api.h"
+
+//Hydra headers
+#include <pxr/base/tf/declarePtrs.h>
+#include <pxr/base/gf/matrix4d.h>
+#include <pxr/imaging/hd/filteringSceneIndex.h>
+
+//The Pixar's namespace needs to be at the highest namespace level for TF_DECLARE_WEAK_AND_REF_PTRS to work.
+PXR_NAMESPACE_OPEN_SCOPE
+
+namespace FVP_NS_DEF {
+
+//We use a filtering scene index to update the data from a HdRetainedSceneIndex where we inserted a parent prim to be the parent of all prims from a data dataProducer scene index
+// which is hosted in a DCC node.
+ 
+class ParentDataModifierSceneIndex;
+TF_DECLARE_WEAK_AND_REF_PTRS(ParentDataModifierSceneIndex);
+
+class ParentDataModifierSceneIndex : public HdSingleInputFilteringSceneIndexBase
```

Comment:
are we expecting this SceneIndex to be used by our clients or just an internal thing ?

Reply:
It's just internal code, not available from the API for customers.